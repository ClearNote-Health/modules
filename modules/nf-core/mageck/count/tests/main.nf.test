// nf-core modules test mageck/count
nextflow_process {

    name "Test Process MAGECK_COUNT"
    script "../main.nf"
    process "MAGECK_COUNT"

    tag "modules"
    tag "modules_nfcore"
    tag "mageck"
    tag "mageck/count"

    test("mus_musculus - fastq") {

       // config "./nextflow.config"

        when {
            process {
                """
                input[0] = [ [ id:'test,test2', single_end:true] , // meta map
                [file(params.test_data['mus_musculus']['illumina']['test_1_fastq_gz'],checkIfExists: true),
                file(params.test_data['mus_musculus']['illumina']['test_2_fastq_gz'], checkIfExists: true)]
                ]
                input[1] = file(params.test_data['mus_musculus']['csv']['library'], checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("mus_musculus - fastq - stub" ) {

        options "-stub"
        config "./nextflow.config"

        when {
            process {
                """
                input[0] = [ [ id:'test,test2', single_end:true] , // meta map
                [file(params.test_data['mus_musculus']['illumina']['test_1_fastq_gz'],checkIfExists: true),
                file(params.test_data['mus_musculus']['illumina']['test_2_fastq_gz'], checkIfExists: true)]
                ]
                input[1] = file(params.test_data['mus_musculus']['csv']['library'], checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match()
                }
            )
        }

    }


}
