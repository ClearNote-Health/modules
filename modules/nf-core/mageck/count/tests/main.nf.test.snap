{
    "mus_musculus - fastq": {
        "content": [
            [
                "versions.yml:md5,8fd54f61f3c77976003f79397f6b12d9"
            ],
            "sgRNA\tGene\ttest\ttest2"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-06-14T14:16:26.720987"
    },
    "mus_musculus - fastq - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test,test2",
                            "single_end": true
                        },
                        "count_table.count_normalized.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test,test2",
                            "single_end": true
                        },
                        "count_table.count_normalized.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,8fd54f61f3c77976003f79397f6b12d9"
                ],
                "count": [
                    [
                        {
                            "id": "test,test2",
                            "single_end": true
                        },
                        "count_table.count_normalized.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "norm": [
                    [
                        {
                            "id": "test,test2",
                            "single_end": true
                        },
                        "count_table.count_normalized.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,8fd54f61f3c77976003f79397f6b12d9"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-06-14T14:16:34.058535"
    }
}